src/PixiEditor/ViewModels/SubViewModels/UpdateViewModel.cs

```diff
@@ -212,9 +213,30 @@ private void Owner_OnClose()
 
     public async Task Download()
     {
-        bool updateCompatible = await UpdateChecker.IsUpdateCompatible();
-        bool updateFileDoesNotExists = !AutoUpdateFileExists();
-        bool updateExeDoesNotExists = !UpdateInstallerFileExists();
+        bool updateCompatible, updateFileDoesNotExists, updateExeDoesNotExists;
+        try
+        {
+            updateCompatible = await UpdateChecker.IsUpdateCompatible();
+            updateFileDoesNotExists = !AutoUpdateFileExists();
+            updateExeDoesNotExists = !UpdateInstallerFileExists();
+        }
+        catch (SocketException ex)
+        {
+            UpdateState = UpdateState.UnableToCheck;
+            return;
+        }
+        catch (IOException ex)
+        {
+            UpdateState = UpdateState.UnableToCheck;
+            return;
+        }
```

Comment:
You could remove these and just put a type check in the `catch (Exception ...)` one